@grafana/ui: Buttons documentation

jessover9000 opened the pull request:
This is a docs update. Added info about how to use buttons and how to combine different button styles next to each other.

jessover9000 committed:
Add info: when to use which and how to mix buttons

sarlinska commented on packages/grafana-ui/src/components/Button/Button.mdx:
```diff
@@ -7,7 +7,7 @@ import { Button } from './Button';
 ## Primary
 
-Used for "call to action".
+Used for "call to action", i.e. triggering the main action. There should never be more than one on a page. If you need multiple buttons for different actions, decide which action is the most important and make that the primary `Button`. All other `Button` components should be secondary.
```
There can be a case when there is no primary action, in such case one should use secondary for everything.

tskarhed replied:
Would it make sense to have it default to secondary, so you don't accidentally end up with 10 primary action buttons?

sarlinska commented on packages/grafana-ui/src/components/Button/Button.mdx:
```diff
@@ -25,7 +25,7 @@ Used for "call to action".
 ## Secondary
 
-The secondary button, used for "cancel" or aborting.
+The secondary `Button` can trigger various actions. When next to the primary `Button`, it can for example be used for "Cancel" or "Abort" actions. Apart from that, the secondary variant is the main `Button` variant in Grafana and is used for most buttons except when there is an important main action.
```
The last sentence is a bit hard to understand. Maybe breaking the whole paragraph into two cases:
1.  When next to the primary `Button`, it can for example be used for "Cancel" or "Abort" actions.
2. When there is no main important action on a given page, the secondary button is used a default primary one.

sarlinska commented on packages/grafana-ui/src/components/Button/Button.mdx:
```diff
@@ -43,7 +43,7 @@ The secondary button, used for "cancel" or aborting.
 ## Destructive
 
-Used for removing or deleting entities.
+Used for "call to action", i.e. triggering the main action, when that main action is removing or deleting something. Because of its dominant coloring, it should be used sparingly. If you need multiple Destrucutive `Button` components in one view, maybe use a secondary `Button` or Link variant instead and only use the Destructive variant to double confirm. Just like the primary variant, the Destru ...
```
1. "call to action" isn't usually associated with destructive actions by people. Maybe we could simplify and say:
"used for triggering action which is removing or deleting something."
2. "maybe use a secondary `Button`..." –> "we recommend using a secondary `Button`..."
3. Not sure the last sentence is needed here.

tskarhed commented:
@jessover9000 Do you want to do something with the feedback?

jessover9000 committed:
Update description for Button.mdx

- add note about scenario without primary action
- add note about secondary button being the default

jessover9000 committed:
Refine docs info for secondary and destructive

tskarhed committed:
Merge branch 'master' into jessover9000/docs-button
